docs/docs/integrations/scoring-reranking-models/3-vertex-ai-ranking.md

```diff
@@ -0,0 +1,90 @@
+---
+sidebar_position: 3
+---
+
+# Google Cloud Vertex AI Ranking API
+
+- [Google Cloud Vertex AI Ranking documentation](https://cloud.google.com/generative-ai-app-builder/docs/ranking)
+- [Google Cloud Vertex AI Ranking API description](https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.rankingConfigs/rank)
+
+
+### Introduction
+
+The Google Cloud Vertex AI Ranking API is a powerful tool that enhances search results by refining the relevance of
+retrieved documents to a given query. Unlike traditional search methods, it leverages advanced machine learning 
+algorithms to understand the semantic context of both the query and the documents, delivering more precise and relevant 
+results. By analyzing the semantic relationship between the query and each document, the API can reorder the candidate 
+documents based on their calculated relevance scores, ensuring that the most relevant results appear at the top of the 
+search results page.
+
+### Maven Dependency
+
+```xml
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-vertex-ai</artifactId>
+    <version>0.34.0</version>
+</dependency>
+```
+
+### Usage
+
+You can score a single string or `TextSegment` against a query 
+thanks to the `score(text, query)` and `score(segment, query)` methods.
+
+It is also possible to score several strings or `TextSegment`s against the query, 
+with the `scoreAll(segments, query)` method:
+
+```java
+VertexAiRankerScoringModel scoringModel = VertexAiRankerScoringModel.builder()
+    .projectId(System.getenv("GCP_PROJECT_ID"))
+    .projectNum(System.getenv("GCP_PROJECT_NUM"))
+    .projectLocation(System.getenv("GCP_LOCATION"))
+    .build();
+
+Response<List<Double>> score = scoringModel.scoreAll(Stream.of(
+        "The sky appears blue due to a phenomenon called Rayleigh scattering. " +
+            "Sunlight is comprised of all the colors of the rainbow. Blue light has shorter " +
+            "wavelengths than other colors, and is thus scattered more easily.",
+
+        "A canvas stretched across the day,\n" +
+            "Where sunlight learns to dance and play.\n" +
+            "Blue, a hue of scattered light,\n" +
+            "A gentle whisper, soft and bright."
+        ).map(TextSegment::from).collect(Collectors.toList()),
+    "Why is the sky blue?");
+
+// [0.8199999928474426, 0.4300000071525574]
+```
+
+By default, the model `semantic-ranker-512@latest` will be used.
```

Comment:
We are moving away from default values for model names (as they get outdated very soon lol), can you please make it mandatory?

Reply:
It's the behavior of the underlying API that I followed.
But ok, I can make it mandatory.

Reply:
Oh, if model name is optional in the API, then let's keep it optional as well (without default value)